### DISCUSSION TITLE
Unable to Uncheck the checkbox  when i am using currentnode.checkbox=false.

### DISCUSSION THREAD

**Discussion by nithyasri123vns** [dowjones/react-dropdown-tree-select#512] [Q&A]
i am trying to implement functionality where user can select only 5 child values per parent in drop down. when user checks 6th value i am showing pop up. But once done with pop up i am trying to uncheck the checkbox which is checked (i.e; 6th item) using checked=false. But its not working .

Here is my code

public onChange = (currentNode, selectedOpt) => {
try {
var a = currentNode
if (a.grandparent == "Brand") {
if (a.checked == true && (Brandcollection.length < 5)) {
Brandcollection.push(a.value)
}

            //if 6th child value is checked after using swal i am unchecking
           else if ((a.checked != false)) {
                Swal.fire({
                    title: 'You can select only 5',
                    icon: 'error',
                    position: 'top'
                });
                a.checked = false
            }
}

But Using this i can uncheck box only once. if again user select the same value, he can see pop up but checkbox is not unchecked? why? please suggest

**Comment by mrchief** [comment]
You can't manipulate the checkboxes directly that way. What you would need to do is re-render the dropdown with the first 5 nodes as `checked: true`.
